Modules/.AL-Go/CompileAppInBcContainer.ps1

```diff
@@ -2,40 +2,42 @@ Param(
     [Hashtable] $parameters
 )
 
+# Setup compiler features to generate captions and LCGs
 if (!$parameters.ContainsKey("Features")) {
     $parameters["Features"] = @()
 }
-$parameters["Features"] = @("lcgtranslationfile", "generateCaptions")
+$parameters["Features"] += @("lcgtranslationfile", "generateCaptions")
 
 $appFile = Compile-AppInBcContainer @parameters
 
-if ($appFile) {
-    $filename = [System.IO.Path]::GetFileName($appFile)
-    if ($filename -like "Microsoft_System Application_*.*.*.*.app") {
-        # System application compiled - add BaseApp and Application app from container to output
-        Invoke-ScriptInBcContainer -containerName $parameters.ContainerName -scriptblock { Param([string]$packagesFolder)
-            function CopyApp {
-                Param(
-                    [string] $localAppPath,
-                    [string] $w1AppPath
-                )
-
-                $name = [System.IO.Path]::GetFileName($w1AppPath)
-                Write-Host "Copying $name to packages path"
-                if (Test-Path $localAppPath) {
-                    $appPath = $localAppPath
-                }
-                else {
-                    $appPath = $w1AppPath
-                }
-                Copy-Item -Path $appPath -Destination (Join-Path $packagesFolder $name)
-                Copy-Item -Path $appPath -Destination (Join-Path "c:\run\my" $name)
-            }
-
-            CopyApp -localAppPath "C:\Applications.*\Microsoft_Base Application_*.*.*.*.app"    -w1AppPath "C:\Applications\BaseApp\Source\Microsoft_Base Application.app"
-            CopyApp -localAppPath "C:\Applications.*\Microsoft_Application_*.*.*.*.app"         -w1AppPath "C:\Applications\Application\Source\Microsoft_Application.app" 
-        } -argumentList (Get-BcContainerPath -ContainerName $parameters.ContainerName -path $Parameters.appSymbolsFolder)
+$branchName = $ENV:GITHUB_REF_NAME
+
+# Only add the source code to the build artifacts if the delivering is allowed on the branch 
+if(($branchName -eq 'main') -or $branchName.StartsWith('release/')) {
+    $appProjectFolder = $parameters.appProjectFolder
+
+    # Extract app name from app.json
+    $appName = (gci -Path $appProjectFolder -Filter "app.json" | Get-Content | ConvertFrom-Json).name
+
+    Write-Host "Current app name: $appName; app folder: $appProjectFolder"
+
+    # TODO Use Al-Go project settings to determine if the app is a test app or not.
+    $holderFolder = 'TestApps'
```

Comment:
Just had a look - there is actually a boolean variable called $app, which is true if this is an app.
There is a variable called $testApp, which is true is this is a test App.

If we make major restructuring of Run-AlPipeline, then this might change, but we have absolutely no plans of doing this.